src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -4310,6 +4309,41 @@ GenerateSyntheticShardIntervalArray(int partitionCount)
 }
 
 
+/*
+ * Determine RowModify required for given query
+ */
+RowModify
+RowModifyForQuery(Query *query)
+{
+	switch (query->commandType)
+	{
+		case CMD_SELECT:
+		{
+			return ROW_MODIFY_READONLY;
+		}
+
+		case CMD_INSERT:
+		{
+			if (query->onConflict != NULL)
```

Comment:
I'm a bit confused here. Why is an `INSERT ... ON CONFLICT ...` is commutative? Am I misunderstanding this condition?

Reply:
Right, originally the condition was `== NULL` & then in a daze looking into why things weren't working I read it thinking "That's not right" & flipped to `!= NULL`